src/array/cpu/csr_sort.cc

```diff
@@ -114,8 +114,8 @@ std::pair<CSRMatrix, NDArray> CSRSortByTag(
       std::vector<IdType> pointer(num_tags, 0);
 
       for (IdType ptr = start ; ptr < end ; ++ptr) {
-        const IdType dst = indices_data[ptr];
-        const TagType tag = tag_data[dst];
+        const IdType eid = eid_data[ptr];
```

Comment:
Try avoid creating `eid_array` at L90-91. Assign `eid_data` pointer to null if `aten::CSRHasData(csr)` is false. Then, here you can just use `const IdType eid = eid_data? eid_data[ptr] : ptr;`.